SCClassLibrary/JITLib/ProxySpace/NodeProxy.sc

```diff
@@ -896,19 +896,26 @@ NodeProxy : BusPlug {
 	// allocation
 
 	freeBus {
-		var oldBus = bus, c;
+		var oldBus = bus;
 		if(oldBus.isNil) { ^this };
-		if(this.isPlaying) {
-			c = (clock ? TempoClock.default);
-			c.sched(server.latency ? 0.01 + quant.nextTimeOnGrid(c) + this.fadeTime, { oldBus.free(true); nil });
-			CmdPeriod.doOnce { if(oldBus.index.notNil) { oldBus.free(true) } };
-		} {
-			oldBus.free(true)
-		};
+		this.schedAfterFade({
+			if(oldBus.index.notNil) { oldBus.free(true) }
+		});
 		busArg = bus = nil;
 		busLoaded = false;
 	}
 
+	schedAfterFade { |func|
+		var delta, usedClock;
+		if(this.isPlaying, {
+			usedClock = (clock ? TempoClock.default);
+			delta = server.latency ? 0.01 + this.fadeTime;
+			if(quant.notNil) { delta = delta + quant.nextTimeOnGrid(usedClock) };
+			usedClock.schedAbs(delta, { func.value; func = nil });
```

Comment:
In line 912, `delta` is a true delta (latency + fade time); in line 913, it changes to become an absolute time.

If `quant` is nil, then with this change, the function will be scheduled for a small time value long in the past, and fire immediately (before the fade has finished).

I'd suggest an else branch `delta = delta + usedClock.beats;` so that after the if, delta is really always an absolute time.

Reply:
ah yes - this has slipped through the tests, I've added some more.

Reply:
strange, do you see:
```
delta = delta + if(quant.notNil) {
				quant.nextTimeOnGrid(usedClock)
			} {
				usedClock.beats
			};
```
?

Reply:
You have to look at the "Commits" page, I think. It looks like the conversation retains the version for which the comment was opened.

It occurred to me later: `nextTimeOnGrid` for quant = 0 *is* clock.beats -- so you can do without the if (btw I've lost my taste for `if` in an expression; yes, we "can" do it and it emphasizes that it's a method rather than a control structure, but after some years, it now seems to me to obscure the intent) and simply `(quant ? 0).nextTimeOnGrid(usedClock)`. I wish I'd thought of that the other day :D

Reply:
ah good. I still like the if as an expression because it emphasizes that something meaningful is returned. But well …